azurerm/internal/services/frontdoor/frontdoor_resource.go

```diff
@@ -223,8 +224,18 @@ func resourceFrontDoor() *pluginsdk.Resource {
 										ValidateFunc: validation.StringInSlice([]string{
 											string(frontdoor.StripAll),
 											string(frontdoor.StripNone),
+											string(frontdoor.StripAllExcept),
 										}, false),
 									},
+									"cache_query_parameters": {
+										Type:     pluginsdk.TypeString,
+										Optional: true,
+									},
+									"cache_duration": {
+										Type:         pluginsdk.TypeString,
+										ValidateFunc: validate.ISO8601DurationBetween("PT1S", "P365D"),
+										Optional:     true,
+									},
```

Comment:
since `cache_query_parameters` is only valid if `cache_query_parameter_strip_directive` is set to `StripAllExcept` can we add a validation check for this condition in the `frontDoorCustomizeDiff`?
same with `cache_duration` can we add a validation check in the `frontDoorCustomizeDiff` to check to see if the `cache_duration` has a value make sure that the `cache_enabled` is set to `true`?

Reply:
Thank you very much @WodansSon! Do you've an example in one of the other resources how this can look like? I've experimented a bit with `RequiredWith` and `ConflictsWith` based on what i found in  the `kubernetes_cluster_resource.go` but wasn't able to make it work.

Reply:
@WodansSon it took me a bit to understand your request. I've added a couple of tests to the `customizediff.go` file now to verify to check the various conditions mentioned above and in the docs. I'm still testing the various conditions and will push my changes in a bit.